keps/sig-scheduling/4328-node-affinity-based-eviction/README.md

```diff
@@ -0,0 +1,506 @@
+- Verify that the daemonSet pod on that node is evicted and not recreated.
+- Add the node's label back.
+- Verify that the daemonSet pod runs on the same node again.
+  
+- Re-start kube-controller-manager, kube-scheduler and kube-apiserver with feature gate disabled
+- Verify the old `.spec.affinity.nodeAffinity.requiredDuringSchedulingRequiredDuringExecution` reserved
+- Remove the node label 
+- Verify that the pod is not evicted
+- Add the node's label back 
+
+- Re-start kube-controller-manager, kube-scheduler and kube-apiserver with feature gate enabled
+- Remove the node label
+- Verify that the daemonSet pod on that node is evicted and not recreated.
+
+###### Is the rollout accompanied by any deprecations and/or removals of features, APIs, fields of API types, flags, etc.?
+
+No.
+
+### Monitoring Requirements
+
+###### How can an operator determine if the feature is in use by workloads?
+
+- Check the corresponding metrics(details in SLIs part below).
+- Inspect the `.spec.affinity.nodeAffinity.requiredDuringSchedulingRequiredDuringExecution` configuration.
+
+###### How can someone using this feature know that it is working for their instance?
+
+- [x] Events
+  - Event Reason: NodeAffinityEviction
+- [x] API .status
+  - Condition reason: DeletionByNodeAffinityEvictionController 
+- [x] Other (treat as last resort)
+  - Details: Pod with `RequiredDuringSchedulingRequiredDuringExecution` must meet affinity rule during both the scheduling phase and the execution phase.
+
+###### What are the reasonable SLOs (Service Level Objectives) for the enhancement?
+
+N/A
+
+###### What are the SLIs (Service Level Indicators) an operator can use to determine the health of the service?
+
+- [x] Metrics
+  - Metric name:
+    - evictions_node_affinity_total(counts of the eviction/evacuation subresources created by the controller)
+    - pod_failed_eviction_node_affinity_total(counts of the pods that failed to be evicted, labeled by pod name)
+###### Are there any missing metrics that would be useful to have to improve observability of this feature?
+
+Two new metric `evictions_node_affinity_total` and `pod_failed_eviction_node_affinity_total` will be added.
+
+### Dependencies
+
+N/A
+
+###### Does this feature depend on any specific services running in the cluster?
+
+No.
+
+### Scalability
+
+###### Will enabling / using this feature result in any new API calls?
+
+- There will be calls to evict pods with eviction subresource. 
+
+###### Will enabling / using this feature result in introducing new API types?
+
+No.
+
+###### Will enabling / using this feature result in any new calls to the cloud provider?
+
+No.
+
+###### Will enabling / using this feature result in increasing size or count of the existing API objects?
+
+- No to existing API objects that doesn't use this feature.
+- For API objects that use this feature:
+  - API type: Pod
+  - Estimated increase in size: new field `.spec.affinity.nodeAffinity.` about 72 bytes.
```

Comment:
Expressing it by a number is probably not enough, and needs a more verbose explanation of what is the expected increase, since we can have multiple NodeSelectorTerm

Reply:
Added an explanation.